lib/ide-haskell-repl.coffee

```diff
@@ -157,3 +157,14 @@ module.exports = IdeHaskellRepl =
         autoScroll: true
 
     upiDisposables
+
+  autocompleteProvider_3_0_0: ->
+    scopeSelector: '.source.haskell'
+    disableForScopeSelector: '.source.haskell .comment'
+    getTextEditorSelector: ->
+      'atom-text-editor.ide-haskell-repl'
+    inclusionPriority: 0
+    getSuggestions: ({editor, prefix}) =>
+      return [] unless @editorMap
+      view = @editorMap.get editor
```

Comment:
While it's a reasonable assumption that `@editorMap` has a relevant editor, we can't guarantee it, since GC could potentially kick in somewhere between callbacks. So I would check if `view` is defined just to be on the safe side.